### PR TITLE
implementing push_transactions api #243

### PR DESCRIPTION
**Pull request by bytemaster**

### CODE REVIEW

File: plugins/chain_plugin/chain_plugin.cpp [hunk 1]
```diff
@@ -263,9 +263,23 @@ read_write::push_block_results read_write::push_block(const read_write::push_blo
...
-   auto ptrx = db.push_transaction(params, skip_flags);
+   auto pretty_input = db.transaction_from_variant( params );
+   auto ptrx = db.push_transaction(pretty_input, skip_flags);
    auto pretty_trx = db.transaction_to_variant( ptrx );
-   return read_write::push_transaction_results{ params.id(), pretty_trx };
+   return read_write::push_transaction_results{ pretty_input.id(), pretty_trx };
+}
+
+read_write::push_transactions_results read_write::push_transactions(const vector<read_write::push_transaction_params>& params) {
+   push_transactions_results result;
```
Comment (heifner) [inline plugins/chain_plugin/chain_plugin.cpp]: Could reserve `result.reserve(params.size());`

File: plugins/chain_plugin/chain_plugin.cpp [hunk 2]
```diff
@@ -263,9 +263,23 @@ read_write::push_block_results read_write::push_block(const read_write::push_blo
...
+   for( const auto& item : params ) {
+      try {
+        result.push_back( push_transaction( item ) ); 
+      } catch ( const fc::exception& e ) {
+        result.push_back( read_write::push_transaction_results{ chain::transaction_id_type(), 
+                          fc::mutable_variant_object( "error", e.to_detail_string() ) } );
+      }
```
Comment (heifner) [inline plugins/chain_plugin/chain_plugin.cpp]: Could use emplace_back()

File: plugins/chain_plugin/chain_plugin.cpp [hunk 3]
```diff
@@ -263,9 +263,23 @@ read_write::push_block_results read_write::push_block(const read_write::push_blo
...
 read_write::push_transaction_results read_write::push_transaction(const read_write::push_transaction_params& params) {
-   auto ptrx = db.push_transaction(params, skip_flags);
+   auto pretty_input = db.transaction_from_variant( params );
+   auto ptrx = db.push_transaction(pretty_input, skip_flags);
    auto pretty_trx = db.transaction_to_variant( ptrx );
-   return read_write::push_transaction_results{ params.id(), pretty_trx };
+   return read_write::push_transaction_results{ pretty_input.id(), pretty_trx };
+}
+
+read_write::push_transactions_results read_write::push_transactions(const vector<read_write::push_transaction_params>& params) {
```
Comment (heifner) [inline plugins/chain_plugin/chain_plugin.cpp]: `const push_transactions_results& params` would match the declaration.

Comment (wanderingbort) [reply plugins/chain_plugin/chain_plugin.cpp]: I think you mean `push_transaction_params` but otherwise 👍

Comment (heifner) [reply plugins/chain_plugin/chain_plugin.cpp]: yes, of course, thanks

**Pull request merged**